src/pages/scripting/samples/index.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Samples
+description: UXP Script samples
+contributors:
+  - https://github.com/amandahuarng
+---
+
+# Samples
+
+## Create a new layer 
+You can create a script that creates a new layer, with the help of the Photoshop DOM API: 
+```js
+const app = require('photoshop').app;
+await app.documents.add();
+await app.activeDocument.createLayer();
+app.activeDocument.layers[0].name = 'New layer';
+```
+
+## Access the local filesystem
+You can create a script that accesses your local filesystem, with the help of the UXP storage module: 
+```js
+const uxpfs = require("uxp").storage;
+const ufs = uxpfs.localFileSystem;
+
+await ufs.getTemporaryFolder().
+    then((result) => {
+        return result.getMetadata();
+    }).then((result) => {
+        console.log(JSON.stringify(result));
+    }).catch((error) => {
+        console.log("Local File system Error: " + error);
+});
+```
+
+## Using a file picker
+You can create a script that instantiates a file picker for writing/saving to files.
+```js
+const fs = require('uxp').storage.localFileSystem;
+try {
+    // save file to a location using file picker
+    const file = await fs.getFileForSaving("demo.txt");
+    await file.write("Hello World! This is demo.");
+} catch (e) {
+    console.log(e);
+}
+```
+
+## Read/write to clipboard
+You can access the clipboard module (`navigator.clipboard`) to: 
+* write to a clipboard (`setContent()`)
+* read a clipboard's contents (`readText()`)
+```js
+const script = await require("uxp").script;
+ 
+scriptResult = "";
+ 
+await testClipboardModuleAccess();
+ 
+script.setResult(scriptResult);
+ 
+async function testClipboardModuleAccess() {
+    return new Promise((resolve, reject) => {
+        try {
+            const clipboard = navigator.clipboard;
+            clipboard.setContent({ 'text/plain': "Test string to validate clipboard setContent" }).then(
+                () => {
+                    console.log("Clipboard write Access Success \n");
+                },
+                (error) => {
+                    console.log("Clipboard write Failed : " + error);
+                    scriptResult += " Clipboard Module Access Failed \n";
+                }
+            );
+            clipboard.readText().then(
+                (result) => {
+                    scriptResult += " Clipboard Module read Access Success \n";
+                    console.log("Clipboard read Success : " + result["text/plain"] + "\n");
+                },
+                (error) => {
+                    console.log("Clipboard read Failed : " + error);
+                    scriptResult += " Clipboard Module read Access Failed \n";
+                }
+            ).finally(() => {
+                resolve();
+            });
+        }  catch (e) {
+            scriptResult += " Clipboard Module Access Failed \n";
+            console.log(e);
+            resolve(e);
+        }
+    })
+}
```

Comment:
This example seems to be unnecessarily complicated. It could/should be much simpler.

Reply:
I was thinking more in favor of async/await. What I seen above reminds me something for unit testing where it can make sense. But devs will be more interested in different use case. 

```
try {
  const clipboard = navigator.clipboard;
  await clipboard.setContent({ 'text/plain': "Test string" });
}catch (e) {
  throw new Error(e);
}
```

```
try {
  const clipboard = navigator.clipboard;
  const text = await clipboard.readText();
  console.log(text);
}catch (e) {
  throw new Error(e);
}
```

I think with top-level await this should work... I don't see a reason to make complicated examples.